### PR TITLE
chore(ci): prefer `working-directory` instead of `cd`

### PR DESCRIPTION
**Pull request by sudosubin**
This PR includes the following changes:

- Features
  - Use `working-directory` instead of `cd` in GitHub Workflows ([docs](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsworking-directory))
  - Update to include GitHub Workflow files in `paths` itself to trigger the lint
  - Add missing `go.mod` path to `.github/workflows/go-lint.yml`
- Display
  - Add missing `name` attribute for GitHub Workflow jobs
  - Update pipeline name (`Csharp Lint` -> `C# Lint`)
  - Update pipeline name (`Javascript` -> `JavaScript`)
- Some formattings (empty space, quote, etc)

### CODE REVIEW

File: .github/workflows/csharp-release.yml [hunk 1]
```diff
@@ -20,19 +20,19 @@ jobs:
...
         run: |
-          cd libraries/csharp
           dotnet restore
+        working-directory: libraries/csharp
 
       - name: Build
         run: |
-          cd libraries/csharp
-          dotnet build --configuration Release StandardWebhooks --no-restore
+          otnet build --configuration Release StandardWebhooks --no-restore
```
Comment (zekth) [inline .github/workflows/csharp-release.yml:28]:
```suggestion
          dotnet build --configuration Release StandardWebhooks --no-restore
```

**Review by zekth — CHANGES_REQUESTED** [review]
one typo slipped in

thanks for this

File: .github/workflows/csharp-release.yml (hunk 1 above)
Comment (sudosubin) [reply .github/workflows/csharp-release.yml]: Oh, so thank you, it was my mistake 😢 🙏

**Comment by zekth** [comment]
seems triggers are wrong for most of the workflows. Would you mind adjusting them @sudosubin ? 
ref: https://github.com/standard-webhooks/standard-webhooks/blob/10f531428d74f502a341d6e73ed2f3490df7679c/.github/workflows/go-lint.yml#L3-L6

**Comment by sudosubin** [comment]
> seems triggers are wrong for most of the workflows. Would you mind adjusting them @sudosubin ? ref:
> 
> https://github.com/standard-webhooks/standard-webhooks/blob/10f531428d74f502a341d6e73ed2f3490df7679c/.github/workflows/go-lint.yml#L3-L6

Of course, but I'm asking to make sure I understand correctly.

Since the workflow pipeline file itself is missing from the `paths` where most lint workflows, so should I add `.github/workflows/{proper-file-name}` for several workflow files?

**Comment by zekth** [comment]
Exactly :)

**Comment by sudosubin** [comment]
It's nice that I could contribute. I reflected it in the code and also modified the PR body. :)

**Review by zekth — APPROVED** [review]
Thanks!

**Pull request merged**